### DISCUSSION TITLE
Meta: How do I make animated GIFs for Tips? (Hints inside)

### DISCUSSION THREAD

**Discussion by aaaaalbert** [pd-externals/tipoftheday-plugin#42] [Q&A]
There is no "official guide" for creating screen capture GIFs -- and I understand that it would be difficult to produce one that caters to the many OSses and window managers etc. out there. However, I keep forgetting and reinventing the required steps on my system, so I document my approach here.

# Platform
* OS: Ubuntu 22.04.2 LTS
* CPU: x86_64 (64-bit AMD)
* Display server: Wayland
* Windowing system: Gnome version 42.5
* Screen recorder: EasyScreenCast version 47 (1.7.1)

# Prerequisites
* [`key-mon`](https://github.com/scottkirkwood/key-mon) (needs Python 3) -- this program opens a little window that displays when the mouse is clicked, modifier keys (<kbd>Shift :arrow_up:</kbd>, <kbd>Ctrl</kbd>, <kbd>Alt</kbd>) and other keys are pressed or held
* [EasyScreenCast](https://github.com/EasyScreenCast/EasyScreenCast) -- records the screen, or selected parts of it, into a video file
* [`ffmpeg`](https://ffmpeg.org/) -- command-line video tool, used here for MP4-to-GIF conversion

# Steps to record screen interactions
* Set up Pure Data as required
* Start `key-mon` and position its window appropriately
* Set up EasyScreenCast for a recording area at most 550 pixels wide (see [GIF format tip](https://github.com/pd-externals/tipoftheday-plugin#gif-format-for-tips))
* Start recording in EasyScreenCast
* *Do your thing*
* Stop recording in EasyScreenCast

# Convert MP4 to GIF
Finally, use `ffmpeg` to convert EasyScreenCast's output file from MP4 format to GIF. The pipeline explained here on [StackExchange superuser](https://superuser.com/questions/556029/how-do-i-convert-a-video-to-gif-using-ffmpeg-with-reasonable-quality) works fine:

```
ffmpeg -ss 30 -t 3 -i input.mp4 \
    -vf "fps=10,scale=320:-1:flags=lanczos,split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse" \
    -loop 0 output.gif
```
**Note well:** Adjust `ss` and `t` to the desired start timestamp and length in your video, set `scale` to the actual width of your recording, and use appropriate input/output file names.

# Done!
Now write a great description and send a [Pull Request](https://github.com/pd-externals/tipoftheday-plugin/pulls) for your new Tip Of The Day!

**Comment by umlaeute** [comment]
thanks for the writeup.

just for the record: i personally found `byzanz` to be a nice/small/simple screenrecorder utility that records directly to GIF
